### ISSUE TITLE
Strange issue with CUDA OOM vs. CPU for fine tune training

### ISSUE THREAD

**Issue by pablogranolabar** [ThilinaRajapakse/simpletransformers#1345]
I've got a fine tune training pipeline going for a multilabel semantic similarity task using T5-large, and when using cuda=False on a 16GB VM I am able to get probably a 6 batch size with 8 or 16 eval batch size (max_seq_length=512 and max_source_length=512).

however on an RTX 3080-class GPU with 16GB of contiguous memory I am getting OOM when simply cuda=True which doesn't make any sense. I've got way more overhead on this VM than the GPU, perhaps gradient checkpointing isn't being used with the GPU implementation?
